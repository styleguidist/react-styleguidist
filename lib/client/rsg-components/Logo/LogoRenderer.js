import React from 'react';
import Styled from 'rsg-components/Styled';

var styles = function styles(_ref) {
  var color = _ref.color,
      fontFamily = _ref.fontFamily,
      fontSize = _ref.fontSize;
  return {
    logo: {
      color: color.base,
      margin: 0,
      fontFamily: fontFamily.base,
      fontSize: fontSize.h4,
      fontWeight: 'normal'
    }
  };
};

export var LogoRenderer = function LogoRenderer(_ref2) {
  var classes = _ref2.classes,
      children = _ref2.children;
  return /*#__PURE__*/React.createElement("h1", {
    className: classes.logo
  }, children);
};
export default Styled(styles)(LogoRenderer);