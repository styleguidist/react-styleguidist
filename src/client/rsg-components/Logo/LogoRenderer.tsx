import React from 'react';
import Styled, { JssInjectedProps } from 'rsg-components/Styled';
import * as Rsg from '../../../typings';

const styles = ({ color, fontFamily, fontSize }: Rsg.Theme) => ({
	logo: {
		color: color.base,
		margin: 0,
		fontFamily: fontFamily.base,
		fontSize: fontSize.h4,
		fontWeight: 'normal',
	},
});

interface Props extends JssInjectedProps {
	children?: React.ReactNode;
}

export const LogoRenderer = ({ classes, children }: Props) => {
	return <h1 className={classes.logo}>{children}</h1>;
};

export default Styled<JssInjectedProps>(styles)(LogoRenderer);
