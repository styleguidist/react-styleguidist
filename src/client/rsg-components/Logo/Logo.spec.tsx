import React from 'react';
import renderer from 'react-test-renderer';
import LogoRenderer from './LogoRenderer';

it('renderer should render header', () => {
	const actual = renderer.create(<LogoRenderer>React Styleguidist</LogoRenderer>);

	expect(actual.toJSON()).toMatchSnapshot();
});
