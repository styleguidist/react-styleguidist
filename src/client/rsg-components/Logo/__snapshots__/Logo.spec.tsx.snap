// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should render header 1`] = `
<h1
  className="rsg--logo-0"
>
  React Styleguidist
</h1>
`;
